packages/cli/test/integration/workflows/workflow-index.test.ts

```diff
@@ -0,0 +1,125 @@
+import { testDb } from '@n8n/backend-test-utils';
+import { Logger } from '@n8n/backend-common';
+import { WorkflowDependencyRepository, WorkflowRepository } from '@n8n/db';
+import { Container } from '@n8n/di';
+import { ErrorReporter } from 'n8n-core';
+import { v4 as uuid } from 'uuid';
+
+import { EventService } from '@/events/event.service';
+import { WorkflowIndexService } from '@/modules/workflow-index/workflow-index.service';
+
+import { createOwner } from '../shared/db/users';
+import { DatabaseConfig } from '@n8n/config';
+
+let workflowIndexService: WorkflowIndexService;
+let eventService: EventService;
+let workflowRepository: WorkflowRepository;
+let workflowDependencyRepository: WorkflowDependencyRepository;
+
+beforeAll(async () => {
+	await testDb.init();
+
+	// Get real instances from the container
+	workflowRepository = Container.get(WorkflowRepository);
+	workflowDependencyRepository = Container.get(WorkflowDependencyRepository);
+	eventService = Container.get(EventService);
+
+	// Create the WorkflowIndexService with real dependencies
+	workflowIndexService = new WorkflowIndexService(
+		workflowDependencyRepository,
+		workflowRepository,
+		eventService,
+		Container.get(Logger),
+		Container.get(ErrorReporter),
+	);
+
+	// Initialize the service to register event listeners
+	workflowIndexService.init();
+});
+
+afterEach(async () => {
+	await testDb.truncate(['WorkflowEntity', 'WorkflowDependency']);
+});
+
+afterAll(async () => {
+	await testDb.terminate();
+});
+
+// NOTE: this feature isn't supported on legacy SQLite databases, so we skip the tests.
+if (Container.get(DatabaseConfig).isLegacySqlite) {
+	describe('WorkflowIndexService Integration', () => {
+		it('is not supported on legacy SQLite databases', async () => {});
+	});
+} else {
+	// NOTE: this feature isn't supported on legacy SQLite databases, so we skip the tests.
+	describe('WorkflowIndexService Integration', () => {
+		describe('workflow-created event', () => {
+			it('should index a new workflow with a single node', async () => {
+				// Arrange
+				const owner = await createOwner();
+				const workflowId = uuid();
+				const versionId = uuid();
+
+				const workflow = {
+					id: workflowId,
+					name: 'Test Workflow',
+					active: false,
+					versionCounter: 1,
+					versionId,
+					nodes: [
+						{
+							id: 'node-1',
+							name: 'HTTP Request',
+							type: 'n8n-nodes-base.httpRequest',
+							typeVersion: 1,
+							position: [250, 300] as [number, number],
+							parameters: {},
+						},
+					],
+					connections: {},
+					settings: {},
+					triggerCount: 0,
+				};
```

Comment:
nit: Can we add `satisfies IWorkflowBase` (or `IWorkflowDb`?)to make sure this object adhers to the type